pandas/core/generic.py

```diff
@@ -5458,13 +5458,63 @@ def asof(self, where, subset=None):
     # Action Methods
 
     _shared_docs['isna'] = """
+        Detect missing values.
+
         Return a boolean same-sized object indicating if the values are NA.
+        NA values, such as None or :attr:`numpy.NaN`, get mapped to True
+        values.
+        Everything else get mapped to False values. Characters such as empty
+        strings `''` or :attr:`numpy.inf` are not considered NA values
+        (unless you set :attr:`pandas.options.mode.use_inf_as_na` `= True`).
+
+        Returns
+        -------
+        bool of type %(klass)s
```

Comment:
Is this type annotation right? It looks like `bool` is the return variable name (which is not needed).
https://python-sprints.github.io/pandas/guide/pandas_docstring.html#section-3-parameters

Reply:
Not the name

Reply:
A complex type then? So it should be somehting like "dict of int", but it looks like it's inverted.

Reply:
We get a DataFrame or a Series  that contains boolean values. 
It feels like writing `DataFrame of bool`or `Series of bool` may be harder to userstand for new users. Any thoughts?

Reply:
The guidelines specify that complex types should be written like `list of bool`. Maybe you can just say that the return type is `DataFrame` and in the explanation specify that its dtype is bool.

Reply:
I think the type should be just `%(klass)s`

```
%(klass)s
    Each element of the %(klass)s will be a boolean.
```

Reply:
Agreed, it's the closes to the guidelines.